torch/csrc/distributed/c10d/quantization/quantization_utils.h

```diff
@@ -0,0 +1,39 @@
+// (c) Facebook, Inc. and its affiliates. Confidential and proprietary.
+
+#pragma once
+
+#include <ATen/ATen.h>
+
+#include <typeinfo>
+
+inline std::string torch_tensor_device_name(const at::Tensor& ten) {
+  return c10::DeviceTypeName(ten.device().type());
+}
+
+inline bool torch_tensor_on_cpu_check(const at::Tensor& ten) {
```

Comment:
what's the necessary part about using these APIs? shouldn't the check be just directly like `tensor.is_cuda()` or not?